kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/matchers/collections/containExactly.kt

```diff
@@ -3,15 +3,35 @@ package io.kotest.matchers.collections
 import io.kotest.assertions.eq.eq
 import io.kotest.assertions.show.Printed
 import io.kotest.assertions.show.show
-import io.kotest.matchers.*
+import io.kotest.matchers.ComparableMatcherResult
+import io.kotest.matchers.Matcher
+import io.kotest.matchers.MatcherResult
+import io.kotest.matchers.neverNullMatcher
+import io.kotest.matchers.should
+import io.kotest.matchers.shouldNot
 import kotlin.jvm.JvmName
 
+/**
+ * Verifies that this collection contains the sub collections provided in the exact given order.
+ *
+ * So, for example, listOf(1,2,3) contains exactly the sub collections:
+ *
+ * [], [1], [2], [3], [1,2], [2,3] and [1,2,3].
```

Comment:
Suppose this is out of scope for the PR, but I was a bit surprised when reading this comment. I expected `shouldContainExactly` to match if, and only if, the actual collection contains _exactly_(?) the elements in expected. I.e. no more and no less than the given. 

E.g.

```kotlin 
listOf(1, 2, 3, 3) shouldContainExactly listOf(1, 2, 3) -> false
setOf(1, 2, 3, 3) shouldContainExactly listOf(1, 2, 3) -> true 
setOf(1, 2, 3, 3) shouldContainExactly listOf(2, 3) -> false
```

Perhaps it's you might as well be using `x shouldBe y` in that case, but well. 

Just thought I'd throw it out there.

Reply:
Actually I think this is the actual behaviour, so the comment is lying? 

Given the below test, `actual` contains the subcollection defined in `expected`, but also an additional element causing the test to fail?
![bild](https://user-images.githubusercontent.com/316857/136113046-a96ff7ac-3ec8-4f2f-84da-84554e578265.png)